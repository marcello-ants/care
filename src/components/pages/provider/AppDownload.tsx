import QRCode from 'qrcode';
import { IconIllustrationXlargeConfirm, IconLogoIconCareLogo } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';
import { makeStyles, Tab, Tabs } from '@material-ui/core';
import { useState, useEffect, useRef } from 'react';
import { ILayoutProps } from '@/types/layout';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';

// Provider app ids; shouldn't be used in seeker flow
const APP_STORE_ID = '1367528046';
const ANDROID_APP_STORE_ID = 'com.care.android.careview.providerapp';

const useStyles = makeStyles((theme) => ({
  desktop: {
    marginTop: '58px',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'stretch',
    gap: theme.spacing(8),
    '& > div': {
      padding: '75px 0 155px',
    },
    [theme.breakpoints.down('sm')]: {
      flexDirection: 'column',
      gap: 0,
      '& > div': {
        padding: '0 0 75px',
      },
    },
  },
  separator: {
    width: '1px',
    background: theme.palette.grey[300],
    flexShrink: 0,
    [theme.breakpoints.down('sm')]: {
      display: 'none',
    },
  },
  qrCodeContainer: {
    width: '410px',
    flexShrink: 0,
    [theme.breakpoints.down('sm')]: {
      width: 'auto',
      textAlign: 'center',
    },
  },
  desktopHeader: {
    display: 'flex',
    justifyContent: 'space-between',
    padding: theme.spacing(3, 4),
    borderBottom: '3px solid #eee',
    '& p': {
      lineHeight: '40px',
    },
  },
  desktopContainer: {
    margin: 'auto',
    maxWidth: '815px',
    [theme.breakpoints.down('sm')]: {
      padding: theme.spacing(0, 3),
    },
  },
  title: {
    textAlign: 'center',
    padding: '25px 0 15px',
    borderBottom: `4px solid ${theme.palette.grey[200]}`,
  },
  hero: {
    display: 'block',
    margin: theme.spacing(0, 'auto', 3),
  },
  heroText: {
    color: theme.palette.care.navy[600],
    fontSize: '28px',
    lineHeight: '36px',
    textAlign: 'right',
    fontFamily: 'Georgia, sans-serif',
    [theme.breakpoints.down('sm')]: {
      textAlign: 'center',
    },
  },
  title2: {
    marginBottom: theme.spacing(3),
    [theme.breakpoints.down('sm')]: {
      marginBottom: theme.spacing(2),
    },
  },
  tabs: {
    marginTop: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center',
    '& button': {
      width: '187px',
      textTransform: 'none',
    },
  },
  buttons: {
    display: 'flex',
    justifyContent: 'center',
    gap: theme.spacing(3),
    margin: theme.spacing(3, 0, 3),
    '& a': {
      display: 'flex',
    },
  },
  linkButton: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    textDecoration: 'none',
    borderRadius: '100px',
    color: theme.palette.care.white,
    background: theme.palette.care.red[500],
    height: theme.spacing(7),
    fontSize: '21px',
    transition: '300ms',
    '&:hover': {
      background: theme.palette.care.red[400],
    },
    marginTop: theme.spacing(4),
  },
  qrcode: {
    margin: theme.spacing(3, 'auto'),
    display: 'block',
  },
}));

const appStoreURL = `https://apps.apple.com/app/id${APP_STORE_ID}`;
const googlePlayURL = `https://play.google.com/store/apps/details?id=${ANDROID_APP_STORE_ID}`;

const isAndroid = typeof window !== 'undefined' && /Android/.test(window.navigator.userAgent);
const isIos = typeof window !== 'undefined' && /iPhone|iPad|iPod/.test(window.navigator.userAgent);
const isDesktop = !isAndroid && !isIos;

const sendAnalytics = () => {
  AnalyticsHelper.logEvent({
    name: 'Member Enrolled',
    data: {
      enrollment_step: 'provider_app_download',
      enrollment_flow: 'MW VHP Provider Enrollment',
      cta_clicked: 'Download the app',
    },
  });
};

const AppDownload = () => {
  const classes = useStyles();
  const ref = useRef<HTMLCanvasElement>(null);

  const [tab, setTab] = useState(0);

  useEffect(() => {
    if (!ref.current) return;
    QRCode.toCanvas(ref.current, tab ? googlePlayURL : appStoreURL, { margin: 0, version: 5 });
  }, [tab]);

  return isDesktop ? (
    <div className={classes.desktop}>
      <div>
        <IconIllustrationXlargeConfirm className={classes.hero} width={300} />
        <Typography className={classes.heroText}>
          You&apos;re one step closer to finding the perfect job.
        </Typography>
      </div>
      <div className={classes.separator} />
      <div className={classes.qrCodeContainer}>
        <Typography variant="h2" className={classes.title2}>
          Last step: Download the app
        </Typography>
        <Typography variant="body2">
          You need the Care.com Caregiver app to continue—it’s free! Get the app and get hired.
        </Typography>

        <div className={classes.tabs}>
          <Tabs value={tab} centered onChange={(event, newValue) => setTab(newValue)}>
            <Tab label="iPhone" />
            <Tab label="Android" />
          </Tabs>
        </div>

        <div className={classes.buttons}>
          {tab === 0 && (
            <a href={appStoreURL} onClick={sendAnalytics}>
              <img src="/app/enrollment/icon-app-store-black.png" alt="Download on the App Store" />
            </a>
          )}
          {tab === 1 && (
            <a href={googlePlayURL} onClick={sendAnalytics}>
              <img src="/app/enrollment/icon-google-play-black.png" alt="Get it on Google Play" />
            </a>
          )}
        </div>

        <canvas className={classes.qrcode} ref={ref} />
        <Typography align="center" careVariant="body3">
          Scan the QR code to download the app
        </Typography>
      </div>
    </div>
  ) : (
    <>
      <IconIllustrationXlargeConfirm className={classes.hero} width={192} />
      <Typography variant="h2" className={classes.title2}>
        You need to download the Care.com Caregiver app to continue — it’s free!
      </Typography>
      <Typography variant="body2">
        Review jobs, reply to messages and receive notifications on the go.
      </Typography>
      <a
        className={classes.linkButton}
        href={isIos ? appStoreURL : googlePlayURL}
        onClick={sendAnalytics}>
        Download now
      </a>
    </>
  );
};

export const AppDownloadHeader = () => {
  const classes = useStyles();

  return (
    <div className={isDesktop ? classes.desktopHeader : classes.title}>
      {isDesktop && <IconLogoIconCareLogo width={145} height={40} />}
      <Typography variant={isDesktop ? 'body2' : 'h4'}>Last step</Typography>
    </div>
  );
};

export const AppDownloadLayout = isDesktop
  ? ({ children }: ILayoutProps) => {
      const classes = useStyles();

      return <div className={classes.desktopContainer}>{children}</div>;
    }
  : null; // use default layout instead

export default AppDownload;
