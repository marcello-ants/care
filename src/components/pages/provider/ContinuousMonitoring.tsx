/* eslint-disable camelcase */
import { useState } from 'react';
import { Button, makeStyles } from '@material-ui/core';
import { Checkbox, FormControlLabel, Typography } from '@care/react-component-lib';
import {
  ContinuousMonitoringEnrollmentStatus,
  ContinuousMonitoringEnrollmentStatus_continuousMonitoringEnrollmentStatus,
} from '@/__generated__/ContinuousMonitoringEnrollmentStatus';
import { useMutation, useQuery } from '@apollo/client';
import { EnrollInContinuousMonitoring } from '@/__generated__/EnrollInContinuousMonitoring';
import {
  ENROLL_IN_CONTINUOUS_MONITORING,
  GET_CONTINUOUS_MONITORING_ENROLLMENT_STATUS,
} from '@/components/request/ContinuousMonitoringGQL';
import OverlaySpinner from '@/components/OverlaySpinner';
import { CLIENT_FEATURE_FLAGS, CZEN_MHP } from '@/constants';
import { useFeatureFlags } from '@/components/FeatureFlagsContext';
import { useProviderCCState } from '@/components/AppState';

type ContinuousMonitoringProps = {
  uid: string;
};

const useStyles = makeStyles((theme) => ({
  header: {
    fontSize: 21,
    fontWeight: 700,
    marginBottom: theme.spacing(2),
  },
  text: {
    marginBottom: theme.spacing(3),
    lineHeight: '22px',
    [theme.breakpoints.down('xs')]: {
      fontSize: 14,
      lineHeight: '20px',
    },
  },
  checkboxLabel: {
    '& span:nth-child(2)': {
      fontSize: 14,
      transform: 'translateY(-1px)',
      lineHeight: '20px',
    },
  },
  buttonWrapper: {
    marginTop: theme.spacing(3),
    textAlign: 'center',
  },
}));

const ContinuousMonitoring = ({ uid }: ContinuousMonitoringProps) => {
  const classes = useStyles();

  const featureFlags = useFeatureFlags();
  const providerState = useProviderCCState();

  const providerCCFreeGatedExperience =
    featureFlags?.flags[CLIENT_FEATURE_FLAGS.PROVIDER_CC_FREE_GATED_EXPERIENCE]?.value;

  const isFreeGated = providerCCFreeGatedExperience && providerState.freeGated;

  const { loading, data: cmConsentData } = useQuery<ContinuousMonitoringEnrollmentStatus>(
    GET_CONTINUOUS_MONITORING_ENROLLMENT_STATUS,
    {
      variables: {
        memberId: uid,
      },
    }
  );

  const { needsConsent, consentMessagesList } =
    cmConsentData?.continuousMonitoringEnrollmentStatus || {};

  if (!isFreeGated || (!loading && !needsConsent)) {
    window.location.assign(CZEN_MHP);
  }

  const [consentIndex, setConsentIndex] = useState(0);
  const [isCurrentConsentAccepted, setIsCurrentConsentAccepted] = useState(false);

  const currentConsentMessage = consentMessagesList?.[consentIndex]?.message;

  const numberOfConsents = consentMessagesList?.length || 1;

  const [enrollInContinuousMonitoring, { loading: enrollIsLoading }] =
    useMutation<EnrollInContinuousMonitoring>(ENROLL_IN_CONTINUOUS_MONITORING, {
      update: (cache, { data }) => {
        if (
          data &&
          data?.enrollInContinuousMonitoring.__typename === 'EnrollInContinuousMonitoringSuccess'
        ) {
          cache.modify({
            fields: {
              continuousMonitoringEnrollmentStatus(
                existingData: ContinuousMonitoringEnrollmentStatus_continuousMonitoringEnrollmentStatus
              ): ContinuousMonitoringEnrollmentStatus_continuousMonitoringEnrollmentStatus {
                return {
                  ...existingData,
                  needsConsent: false,
                };
              },
            },
          });
        }
      },
    });

  const acceptConsentHandler = async () => {
    const consentGivenMessageIds =
      consentMessagesList &&
      consentMessagesList.map((consent) => consent && consent.consentMessageId);

    enrollInContinuousMonitoring({
      variables: {
        input: {
          consentGivenMessageIds,
          memberId: uid,
        },
      },
    }).then(() => {
      window.location.assign(CZEN_MHP);
    });
  };

  if (!needsConsent || !isFreeGated) return <OverlaySpinner isOpen wrapped />;

  return (
    <>
      <Typography className={classes.header} variant="h3" align="center">
        Disclosure
      </Typography>
      {currentConsentMessage && (
        <div
          className={classes.text}
          dangerouslySetInnerHTML={{
            __html: currentConsentMessage,
          }}
        />
      )}
      <Typography className={classes.header} variant="h3" align="center">
        Authorization
      </Typography>
      <FormControlLabel
        className={classes.checkboxLabel}
        label="By checking this box, I acknowledge that I have read, understand, and agree to this Disclosure, and I authorize the requested background check."
        control={
          <Checkbox
            checked={isCurrentConsentAccepted}
            onChange={(e) => setIsCurrentConsentAccepted(e.target.checked)}
          />
        }
      />
      <div className={classes.buttonWrapper}>
        {consentIndex === numberOfConsents - 1 ? (
          <Button
            color="secondary"
            variant="contained"
            size="medium"
            onClick={acceptConsentHandler}
            disabled={!isCurrentConsentAccepted || enrollIsLoading}>
            Continue
          </Button>
        ) : (
          <Button
            color="secondary"
            variant="contained"
            size="medium"
            disabled={!isCurrentConsentAccepted}
            onClick={() => {
              setConsentIndex(consentIndex + 1);
              setIsCurrentConsentAccepted(false);
            }}>
            Next
          </Button>
        )}
      </div>
    </>
  );
};

export default ContinuousMonitoring;
