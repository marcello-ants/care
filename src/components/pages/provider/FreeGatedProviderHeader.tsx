import React, { useEffect, useRef, useState } from 'react';
import getConfig from 'next/config';
import { createPortal } from 'react-dom';

import { Button, makeStyles } from '@material-ui/core';
import { ProfileAvatar, Typography } from '@care/react-component-lib';
import {
  Icon24UtilityChevronSmall,
  Icon24UtilityClose,
  Icon24UtilityMenu,
  IconLogoCarelogo,
} from '@care/react-icons';

import { getCaregiver_getCaregiver_member as MemberDetails } from '@/__generated__/getCaregiver';

const { publicRuntimeConfig } = getConfig();

const { CZEN_GENERAL } = publicRuntimeConfig;

type StylesProps = {
  isMenuOpen: boolean;
};

const useStyles = makeStyles((theme) => ({
  header: {
    position: 'absolute',
    top: 0,
    left: 0,
    height: theme.spacing(13),
    width: '100%',
    padding: theme.spacing(3, 4),
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderBottom: `1px solid ${theme.palette.care.grey[300]}`,
    [theme.breakpoints.down('sm')]: {
      padding: theme.spacing(2),
      height: theme.spacing(7),
    },
  },
  logo: {
    height: theme.spacing(5),
    width: theme.spacing(18),
    [theme.breakpoints.down('sm')]: {
      height: theme.spacing(4),
      width: 116,
    },
  },
  hamburgerIcon: {
    marginRight: theme.spacing(2),
    width: theme.spacing(4),
    height: theme.spacing(4),
    display: 'none',
    [theme.breakpoints.down('md')]: {
      display: 'block',
    },
    [theme.breakpoints.down('sm')]: {
      width: theme.spacing(3),
      height: theme.spacing(3),
    },
  },
  hamburgerMenuContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  avatarMenuButton: {
    display: 'flex',
    alignItems: 'center',
    cursor: 'pointer',
    background: 'none',
    border: 'none',
    outline: 'none',
    '-webkit-tap-highlight-color': 'transparent',
    [theme.breakpoints.down('md')]: {
      display: 'none',
    },
  },
  chevronIcon: {
    marginLeft: 4,
  },
  mobileMenu: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column',
    top: 80,
    right: 80,
    boxShadow: '0 4px 16px rgba(0, 0, 0, 0.15)',
    width: '100%',
    maxWidth: theme.spacing(40.75),
    padding: theme.spacing(3, 4),
    minHeight: 325,
    zIndex: theme.zIndex.drawer,
    backgroundColor: theme.palette.care.white,
    borderRadius: `0 0 ${theme.spacing(2)}px ${theme.spacing(2)}px`,
    [theme.breakpoints.up('lg')]: {
      display: ({ isMenuOpen }: StylesProps) => (isMenuOpen ? 'flex' : 'none'),
    },
    [theme.breakpoints.down('md')]: {
      padding: theme.spacing(6, 7, 5),
      position: 'fixed',
      left: 0,
      top: 0,
      height: '100%',
      maxWidth: 375,
      borderRadius: 0,
      transform: ({ isMenuOpen }: StylesProps) => !isMenuOpen && 'translateX(-101%)',
      willChange: 'transform',
      transition: 'transform .15s cubic-bezier(0,0,.3,1)',
    },
  },
  mobileMenuCloseIcon: {
    display: 'none',
    position: 'absolute',
    top: theme.spacing(3),
    right: theme.spacing(3),
    cursor: 'pointer',
    [theme.breakpoints.down('md')]: {
      display: 'block',
    },
  },
  profileName: {
    margin: theme.spacing(1, 0, 6),
  },
  menuItemsContainer: {
    width: '100%',
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    flexGrow: 1,
    zIndex: 9999,
  },
  menuHeader: {
    paddingBottom: theme.spacing(1),
    marginBottom: 6,
    borderBottom: `1px solid ${theme.palette.care.navy[800]}`,
  },
  menuLink: {
    padding: '11px 0',
    color: '#172C42',
    width: '100%',
    display: 'block',
    marginBottom: theme.spacing(3),
    textDecoration: 'none',
    '&:hover': {
      color: theme.palette.care.navy[900],
      background: theme.palette.care.grey[200],
    },
  },
}));

type FreeGatedProviderHeaderProps = {
  memberDetails: MemberDetails | null;
  logoutHandler: () => void;
};

const FreeGatedProviderHeader = ({
  memberDetails,
  logoutHandler,
}: FreeGatedProviderHeaderProps) => {
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  const nextRoot = document.getElementById('__next');

  const classes = useStyles({ isMenuOpen });

  const menuRef = useRef<HTMLDivElement>(null);

  const toggleMenuVisibilityHandler = () => {
    setIsMenuOpen(!isMenuOpen);
  };

  const avatarButtonClickHandler = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();

    toggleMenuVisibilityHandler();
  };

  const dismissListener = (event: Event) => {
    if (!menuRef?.current?.contains(event.target as Node)) {
      event.stopPropagation();

      toggleMenuVisibilityHandler();
    }
  };

  useEffect(() => {
    if (isMenuOpen) {
      document.addEventListener('click', dismissListener, true);
    }

    return () => document.removeEventListener('click', dismissListener, true);
  }, [isMenuOpen]);

  if (nextRoot) {
    return createPortal(
      <>
        <div className={classes.header}>
          <div className={classes.hamburgerMenuContainer}>
            <Icon24UtilityMenu
              className={classes.hamburgerIcon}
              onClick={toggleMenuVisibilityHandler}
            />
            <IconLogoCarelogo className={classes.logo} />
          </div>
          <button
            type="button"
            className={classes.avatarMenuButton}
            onClick={avatarButtonClickHandler}>
            <ProfileAvatar size="small" src={memberDetails?.imageURL || undefined} />
            <Icon24UtilityChevronSmall className={classes.chevronIcon} />
          </button>
        </div>
        <div className={classes.mobileMenu} ref={menuRef}>
          <Icon24UtilityClose
            className={classes.mobileMenuCloseIcon}
            onClick={toggleMenuVisibilityHandler}
          />
          <ProfileAvatar size="xLarge" src={memberDetails?.imageURL || undefined} />
          <Typography variant="h3" className={classes.profileName}>
            {memberDetails?.displayName}
          </Typography>
          <div className={classes.menuItemsContainer}>
            <div>
              <Typography variant="h3" className={classes.menuHeader}>
                Account
              </Typography>
              <Typography variant="body1">
                <a href={`${CZEN_GENERAL}/member/profile.do`} className={classes.menuLink}>
                  Settings & Privacy
                </a>
              </Typography>
            </div>
            <Button
              color="secondary"
              variant="outlined"
              size="small"
              fullWidth
              data-testid="logout-button"
              onClick={logoutHandler}>
              Log out
            </Button>
          </div>
        </div>
      </>,
      nextRoot
    );
  }

  return null;
};

export default FreeGatedProviderHeader;
