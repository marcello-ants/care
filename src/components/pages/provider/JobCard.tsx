import { makeStyles } from '@material-ui/core';
import { Card, Divider, ReadMore, Typography } from '@care/react-component-lib';
import { Icon24InfoPayments, Icon24InfoLocation } from '@care/react-icons';

const useStyles = makeStyles((theme) => ({
  header: {
    padding: theme.spacing(0, 2, 0.5),
  },
  jobPostDate: {
    color: theme.palette.care.grey[600],
    margin: theme.spacing(2, 0, 0.5, 2),
  },
  body: {
    padding: theme.spacing(0, 2),
    display: 'inline-block',
    width: '100%',
    // TODO: remove body3's styles after SC-489 is done
    fontWeight: 400,
    fontSize: '0.875rem',
    lineHeight: '20px',
  },
  divider: {
    margin: theme.spacing(1, 0),
    '&::before': {
      borderTopStyle: 'dashed',
    },
  },
  footer: {
    position: 'relative',
    height: '30px',
  },
  footerLocation: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    left: '20px',
  },
  footerRate: {
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    right: '20px',
  },
  icon: {
    paddingRight: theme.spacing(1),
  },
}));

export interface JobCardProps {
  header: string;
  jobPostDate: string;
  body: string;
  location: string;
  rate: string;
}
export default function JobCard({ header, jobPostDate, body, location, rate }: JobCardProps) {
  const classes = useStyles();
  return (
    <Card variant="outlined">
      <Typography careVariant="tag" className={classes.jobPostDate}>
        <span>{jobPostDate}</span>
      </Typography>
      <Typography variant="h4" className={classes.header}>
        <div>{header}</div>
      </Typography>
      <ReadMore
        variant="body2" // TODO: replace with "body3" after SC-489 is done
        showLessButton
        charLimit={80}
        value={body}
        className={classes.body}
      />
      <Divider classes={classes.divider} />
      <div className={classes.footer}>
        <div className={classes.footerLocation}>
          <Icon24InfoLocation size={24} className={classes.icon} />
          <Typography careVariant="body3">
            <span>{location}</span>
          </Typography>
        </div>
        <div className={classes.footerRate}>
          <Icon24InfoPayments size={24} className={classes.icon} />
          <Typography careVariant="body3">
            <span>{rate}</span>
          </Typography>
        </div>
      </div>
    </Card>
  );
}
