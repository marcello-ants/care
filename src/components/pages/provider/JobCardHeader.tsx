import { Grid, makeStyles } from '@material-ui/core';
import Header from '@/components/Header';

const useStyles = makeStyles((theme) => ({
  blueText: {
    color: theme.palette.care.blue[700],
  },
}));
interface JobCardHeaderProps {
  jobsLength: number;
  numberOfJobsNear: number;
}
export default function JobCardHeader({ jobsLength, numberOfJobsNear }: JobCardHeaderProps) {
  const classes = useStyles();

  if (jobsLength === 1)
    return (
      <Grid item xs={12}>
        <Header>
          <span>There is</span>
          <span className={classes.blueText}> 1 new job </span>
          <span>that matches your specified skills.</span>
        </Header>
      </Grid>
    );

  return (
    <Grid item xs={12}>
      <Header>
        <span>Here are</span>
        <span className={classes.blueText}>
          {jobsLength > numberOfJobsNear
            ? ` ${jobsLength} new jobs `
            : ` ${jobsLength} of ${numberOfJobsNear} jobs `}
        </span>
        <span>that match your skills.</span>
      </Header>
    </Grid>
  );
}
