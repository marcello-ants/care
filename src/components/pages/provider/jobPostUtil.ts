import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';
import { AnalyticsHelper } from '@/utilities/analyticsHelper';
import {
  getNewJobsForProvider,
  getNewJobsForProviderVariables,
} from '@/__generated__/getNewJobsForProvider';
import { GET_NEW_JOBS_PROVIDER } from '@/components/request/GQL';
import { useQuery } from '@apollo/client';
import { JobCardProps } from './JobCard';

const numResults = 3;
dayjs.extend(relativeTime);

export function jobPostDate(date1: string, date2: Date) {
  const hoursDiff = dayjs(date2).diff(date1, 'hour');
  let jobPostDesc;
  if (hoursDiff <= 1) {
    jobPostDesc = 'posted 1 hour ago';
  } else if (hoursDiff < 24 * 7) {
    jobPostDesc = `posted ${dayjs(date1).from(date2)}`;
  } else {
    jobPostDesc = '';
  }
  return jobPostDesc;
}
const currencyFormatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 0,
  maximumFractionDigits: 2,
});
export function useJobList() {
  const {
    loading,
    data: jobSummaries,
    error: jobsError,
  } = useQuery<getNewJobsForProvider, getNewJobsForProviderVariables>(GET_NEW_JOBS_PROVIDER, {
    variables: {
      numResults,
    },
  });
  const jobs = jobSummaries?.getNewJobsForProvider.map((jobSummary) => {
    const hourlyRateFrom = currencyFormatter.format(
      Math.ceil(parseFloat(jobSummary.payRange.hourlyRateFrom.amount))
    );
    const jobCardProps: JobCardProps = {
      header: jobSummary.title || '',
      jobPostDate: jobPostDate(jobSummary.jobPostDate, new Date()),
      body: jobSummary.description || '',
      location: `${jobSummary.city}, ${jobSummary.state}`,
      rate: `${hourlyRateFrom}-${jobSummary.payRange.hourlyRateTo.amount}/hr`,
    };
    return jobCardProps;
  });
  return {
    jobs,
    loading,
    jobsError,
  };
}

export function jobsNearAnalytics(numberOfJobsNear: number) {
  AnalyticsHelper.logEvent({
    name: 'Member Enrolled',
    data: {
      enrollment_flow: 'MW VHP Provider Enrollment',
      enrollment_step: 'new_jobs_near_you',
      cta_clicked: 'Next',
      number_of_jobs: numberOfJobsNear,
    },
  });
}
