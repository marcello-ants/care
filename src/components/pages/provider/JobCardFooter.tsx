import React from 'react';
import { Button } from '@material-ui/core';
import { jobsNearAnalytics } from '@/components/pages/provider/jobPostUtil';
import FixElementToBottom from '@/components/FixElementToBottom';
import useEnterKey from '@/components/hooks/useEnterKey';

interface JobCardFooterProps {
  numberOfJobsNear: number;
  nextUrl: string;
}
export default function JobCardFooter({ numberOfJobsNear, nextUrl }: JobCardFooterProps) {
  const handleNext = () => {
    jobsNearAnalytics(numberOfJobsNear);
    window.location.assign(nextUrl);
  };
  useEnterKey(true, handleNext);
  return (
    <FixElementToBottom>
      <Button color="primary" variant="contained" size="large" fullWidth onClick={handleNext}>
        Next
      </Button>
    </FixElementToBottom>
  );
}
