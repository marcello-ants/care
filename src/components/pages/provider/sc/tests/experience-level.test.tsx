import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { useRouter } from 'next/router';
import { MockedProvider } from '@apollo/client/testing';

import { AppStateProvider } from '@/components/AppState';
import { SENIOR_CARE_PROVIDER_ATTRIBUTES_UPDATE } from '@/components/request/GQL';
import { initialAppState } from '@/state';
import ExperienceLevel from '@/pages/provider/sc/experience-level';
import { PROVIDER_ROUTES } from '@/constants';
import { LevelOfExperienceLabels } from '@/types/provider';

const gqlMocks = [
  {
    request: {
      query: SENIOR_CARE_PROVIDER_ATTRIBUTES_UPDATE,
      variables: {
        input: {
          yearsOfExperience: 10,
        },
      },
    },
    result: {
      data: {
        seniorCareProviderAttributesUpdate: {
          __typename: 'SeniorCareProviderAttributesUpdateSuccess',
        },
      },
    },
  },
];

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

describe('Experience level page', () => {
  let mockRouter: any = null;
  let asFragment: any | null = null;

  const renderComponent = () => {
    const view = render(
      <MockedProvider mocks={gqlMocks} addTypename={false}>
        <AppStateProvider initialStateOverride={initialAppState}>
          <ExperienceLevel />
        </AppStateProvider>
      </MockedProvider>
    );
    ({ asFragment } = view);
  };

  beforeEach(() => {
    mockRouter = {
      push: jest.fn(),
      pathname: PROVIDER_ROUTES.EXPERIENCE_LEVEL,
    };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
  });

  afterEach(() => {
    // cleanup on exiting
    mockRouter = null;
    asFragment = null;
  });

  it('matches snapshot', () => {
    renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it('check options and redirect to the next page', async () => {
    renderComponent();
    const option1 = screen.getByText(LevelOfExperienceLabels.YEARS_1);
    const option2 = screen.getByText(LevelOfExperienceLabels.YEARS_3);
    const option3 = screen.getByText(LevelOfExperienceLabels.YEARS_5);
    const option4 = screen.getByText(LevelOfExperienceLabels.YEARS_10);
    const option5 = screen.getByText(LevelOfExperienceLabels.YEARS_0);

    expect(option1).toBeInTheDocument();
    expect(option2).toBeInTheDocument();
    expect(option3).toBeInTheDocument();
    expect(option4).toBeInTheDocument();
    expect(option5).toBeInTheDocument();

    fireEvent.click(option4);

    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(mockRouter.push).toHaveBeenCalledWith(PROVIDER_ROUTES.PROFILE));
  });
});
