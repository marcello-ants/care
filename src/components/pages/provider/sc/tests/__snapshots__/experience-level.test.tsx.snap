// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience level page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-2 MuiTypography-h2"
      >
        How much paid senior caregiving experience do you have?
      </h2>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <ul
        class="MuiList-root makeStyles-selector-1 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-root-4 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-7"
            id="selector.experienceLevel.0"
            name="experienceLevel"
            readonly=""
            type="radio"
            value="1"
          />
          <div
            class="makeStyles-controlIndicator-8 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  1-2 years
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-4 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-7"
            id="selector.experienceLevel.1"
            name="experienceLevel"
            readonly=""
            type="radio"
            value="3"
          />
          <div
            class="makeStyles-controlIndicator-8 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  3-4 years
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-4 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-7"
            id="selector.experienceLevel.2"
            name="experienceLevel"
            readonly=""
            type="radio"
            value="5"
          />
          <div
            class="makeStyles-controlIndicator-8 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  5-9 years
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-4 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-7"
            id="selector.experienceLevel.3"
            name="experienceLevel"
            readonly=""
            type="radio"
            value="10"
          />
          <div
            class="makeStyles-controlIndicator-8 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  10+ years
                </p>
              </div>
            </div>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-root-4 pill MuiListItem-gutters"
        >
          <input
            class="makeStyles-input-7"
            id="selector.experienceLevel.4"
            name="experienceLevel"
            readonly=""
            type="radio"
            value="0"
          />
          <div
            class="makeStyles-controlIndicator-8 md"
            tabindex="0"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center"
            >
              <div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  I am seeking my first senior care job
                </p>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
